app/src/utils/utils.ts

```diff
@@ -1,11 +1,31 @@
-import { ICalendarItem, ICourse } from 'components/shared/interfaces/timetable.interfaces';
+import {
+  ICalendarItem_Course,
+  ICourse,
+  ICalendarItem_Teacher
+} from 'components/shared/interfaces/timetable.interfaces';
+import colors from 'data/CourseColor.json';
 
 /**
  * Grab data from python scraper and format it for DevExtreme Scheduler
  *  Reference: https://js.devexpress.com/Demos/WidgetsGallery/Demo/Scheduler/CustomTemplates/React/Light/
  */
-export function parseCalendarJSON(data: ICourse[]): ICalendarItem[] {
-  const calendarData: ICalendarItem[] = [];
+
+export function parseCalendarJSON_Teacher(data: ICourse[]): ICalendarItem_Teacher[] {
+  const calendarData_Teacher: ICalendarItem_Teacher[] = [];

```

Comment:
I'm not sure I'm following why we are pulling out the professor from the course. I have a couple of questions:
* Since we're doing data.forEach(), are we adding an entry to `calendarData_Teacher` for every section's professor? I think this would mean we'd duplicate the professor if they taught more than one class. If that's the case I don't see a point to pulling out the professor since we'll have a 1:1 mapping anyways. 
* I agree with Ahnaf's point, I would switch the color determination to be set up based on the course number, not prof ID so the same course will have the same color.
* Also agree with the suggestion to add section number (ie A01). You could just append it to the title.

Reply:
The main reason I create another entry for teacher is that we want assign each teacher a color, and it is also used by the assign multiple teachers in one course feature.